documentation/src/docs/getting-started.md

```diff
@@ -0,0 +1,240 @@
+
+1. The first part of the application imports our dependencies, extracts a few environment variables, and obtains a `SparkSession`.
+
+    ```python title="stream_generated_data_to_pravega.py"
+    from pyspark.sql import SparkSession
+    import os
+
+    controller = os.getenv("PRAVEGA_CONTROLLER_URI", "tcp://127.0.0.1:9090")
+    allowCreateScope = os.getenv("PROJECT_NAME") is None
+    scope = os.getenv("PRAVEGA_SCOPE", "spark")
+    checkPointLocation = os.getenv("CHECKPOINT_DIR",
+                                   "/tmp/spark-checkpoints-stream_generated_data_to_pravega")
+
+    spark = (SparkSession
+            .builder
+            .getOrCreate()
+            )
+    ```
+
+2. Next, we read from a [rate source](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#creating-streaming-dataframes-and-streaming-datasets), which continuously generates rows containing a timestamp.
+
+    ```python
+    (spark
+        .readStream
+        .format("rate")
+        .load()
+    ```
+
+3. Next, we will use a SQL expression to define a simple transformation on the rate source output. When writing to Pravega, we will need to provide a string or binary (byte sequence) column named `event`, and optionally a string column named `routing_key`.
+
+    ```python
+        .selectExpr("cast(timestamp as string) as event", "cast(value as string) as routing_key")
+    ```
+
+4. Next, we will specify that we want to write the output every 3 seconds. When writing to Pravega, you will typically want to use the append output mode.
+
+    ```python
+        .writeStream
+        .trigger(processingTime="3 seconds")
+        .outputMode("append")
+    ```
+
+5. Now we specify the target for the output. This will write to the Pravega stream named `streamprocessing1`.
+
+    ```python
+        .format("pravega")
+        .option("allow_create_scope", allowCreateScope)
+        .option("controller", controller)
+        .option("scope", scope)
+        .option("stream", "streamprocessing1")
+        .option("checkpointLocation", checkPointLocation)
+    ```
+
+6. Finally, we start the Spark job that we defined and wait for it to complete. Since this is an unbounded job, it will continue to run until stopped by the user.
+
+    ```python title="stream_generated_data_to_pravega.py"
+        .start()
+        .awaitTermination()
+    )
+    ```
+
+7. To run this application locally and write to your local development installation of Pravega:
+
+    ```shell
+    ./run_pyspark_app.sh src/main/python/stream_generated_data_to_pravega.py
+    ```
+
+   This job will continue to run and write events until stopped.
+
+<IfHaveFeature feature="nautilus">
+
+8. To deploy this Python Spark application on SDP:
+
+   1. [Upload Common Artifacts to your Analytics Project](../sdp/analytics/spark/deploying#upload-common-artifacts-to-your-analytics-project).
```

Comment:
We referenced another doc link which is not in this repo here. I'm a little worried about this as it would be hard to maintain as the reference and the source are not in one place, for example, like the doc directory change will also need a change here.

I know this works with docusaurus as a whole project now, but I would prefer if we need linking to internal docs, we do it in a reverse way, so we can keep this self contained and people won't get confused when reading this markdown file.

Reply:
I rearranged the documentation so that the only reference to SDP is in getting-started.md. It contains:

```
<IfHaveFeature feature="nautilus">
:::tip Streaming Data Platform
If you are using **Dell EMC Streaming Data Platform**, you should use [Getting Started with Spark on SDP](../sdp/analytics/spark/getting-started.md) instead of this guide.
:::
</IfHaveFeature>
```

Once we make the SDP documentation public, we may want to remove the `IfHaveFeature` check.

I think this note is important so that SDP users do not start using the wrong Getting Started guide. What do you think?

Reply:
I rearranged the documentation yet again. It now imports external documentation using this pattern:
```
import DeployPythonSpark from '../snippets/spark-connectors/deploy-python-spark.md';

<DeployPythonSpark />
```
This avoids all references to non-OSS products.